packages/functions/src/__tests__/or.test.ts

```diff
@@ -0,0 +1,237 @@
+      ),
+    ).toEqual([]);
+  });
+
+  it('given none of 5 properties, should return an error message', async () => {
+    expect(
+      await runOr(
+        {
+          version: '1.0.0',
+          title: 'Swagger Petstore',
+          termsOfService: 'http://swagger.io/terms/',
+        },
+        { properties: ['yada-yada', 'foo', 'bar', 'four', 'five'] },
+      ),
+    ).toEqual([
+      {
+        message: 'At least one of "yada-yada" or "foo" or "bar" or 2 other properties must be defined',
+        path: [],
+      },
+    ]);
+  });
+
+  it('given only one of 4 properties, should return no error message', async () => {
+    expect(
+      await runOr(
+        {
+          version: '1.0.0',
+          title: 'Swagger Petstore',
+          termsOfService: 'http://swagger.io/terms/',
+        },
+        { properties: ['title', 'foo', 'bar', 'four'] },
+      ),
+    ).toEqual([]);
+  });
+
+  describe('validation', () => {
+    it.each([{ properties: ['foo', 'bar'] }])('given valid %p options, should not throw', async opts => {
+      expect(await runOr([], opts)).toEqual([]);
+    });
+
+    it.each([{ properties: ['foo'] }])('given valid %p options, should not throw', async opts => {
+      expect(await runOr([], opts)).toEqual([]);
+    });
+
+    it.each([{ properties: ['foo', 'bar', 'three'] }])('given valid %p options, should not throw', async opts => {
+      expect(await runOr([], opts)).toEqual([]);
+    });
+
+    it.each<[unknown, RulesetValidationError[]]>([
+      [
+        null,
+        [
+          new RulesetValidationError(
+            'invalid-function-options',
+            '"or" function has invalid options specified. Example valid options: { "properties": ["default", "example"] }, { "properties": ["title", "summary", "description"] }, etc.',
+            ['rules', 'my-rule', 'then', 'functionOptions'],
+          ),
+        ],
+      ],
+      [
+        2,
+        [
+          new RulesetValidationError(
+            'invalid-function-options',
+            '"or" function has invalid options specified. Example valid options: { "properties": ["default", "example"] }, { "properties": ["title", "summary", "description"] }, etc.',
+            ['rules', 'my-rule', 'then', 'functionOptions'],
+          ),
+        ],
+      ],
+      [
+        { properties: ['foo', 'bar'], foo: true },
+        [
+          new RulesetValidationError('invalid-function-options', '"or" function does not support "foo" option', [
+            'rules',
+            'my-rule',
+            'then',
+            'functionOptions',
+            'foo',
+          ]),
+        ],
+      ],
+      [
+        { properties: ['foo', {}] },
+        [
+          new RulesetValidationError(
+            'invalid-function-options',
+            '"or" requires one or more enumerated "properties", i.e. ["default", "example"], ["title", "summary", "description"], etc.',
```

Comment:
This does not match what will now be returned by the JSON Schema:
'"or" requires at least two enumerated "properties", i.e. ["default", "example"], ["title", "summary", "description"], etc.'